File: app/src/main/java/com/iitk/proxyLogin/SplashActivity.java
package com.iitk.proxyLogin;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.support.v4.content.LocalBroadcastManager;
import android.support.v7.app.AppCompatActivity;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.TextView;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Created by kshivang on 22/01/17.
 *
 */

public class SplashActivity extends AppCompatActivity {

    private boolean isAnimationFinished = false;
    private TextView tvProgress;
    private LocalBroadcastManager localBroadcastManager;

    private final BroadcastReceiver proxyReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            if ("proxy.app.PROXY_PROGRESS".equals(intent.getAction())) {
                tvProgress.setText(intent.getStringExtra("Progress"));
            } else if ("proxy.app.PROXY_REQUEST_CREDENTIAL".equals(intent.getAction())) {
                String type = intent.getStringExtra("Type");
                tvProgress.setText("You are on IITK " + type + " network");
                onFetched(true);
            } else if ("proxy.app.PROXY_LIVE_SESSION".equals(intent.getAction())) {
                long lastLogin = intent.getLongExtra("Time",
                                        System.currentTimeMillis());
                tvProgress.setText("IITK Fortinet refreshed at " +
                        new SimpleDateFormat("HH:mm a", Locale.ENGLISH)
                                .format(new Date(lastLogin)));
                onFetched(false);
            } else {
                tvProgress.setText("You are not on IITK network");
                onFetched(false);
            }
        }
    };


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        localBroadcastManager = LocalBroadcastManager.getInstance(this);
        localBroadcastManager.registerReceiver(proxyReceiver, makeProxyUpdatesIntentFilter());

        fetchType();

        Animation bounceAnim = AnimationUtils
                .loadAnimation(getApplicationContext(), R.anim.bounce);

        tvProgress = (TextView) findViewById(R.id.progress_text);
        TextView tvTitle = (TextView) findViewById(R.id.appName);

        tvTitle.setAnimation(bounceAnim);

        bounceAnim.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {
            }

            @Override
            public void onAnimationEnd(Animation animation) {
                isAnimationFinished = true;
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });
    }

    private void fetchType() {
        Intent proxyServiceIntent = new Intent(this, ProxyService.class);
        proxyServiceIntent.setAction("proxy.service.NETWORK_TYPE");
        startService(proxyServiceIntent);
    }

    private void onFetched(boolean isLoginRequired) {
        startActivity(new Intent(SplashActivity.this, isLoginRequired ?
                LoginActivity.class : SessionActivity.class));
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (this.localBroadcastManager != null) {
            localBroadcastManager.unregisterReceiver(proxyReceiver);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (isAnimationFinished) {
            finish();
        }
    }

    public static IntentFilter makeProxyUpdatesIntentFilter() {
        IntentFilter intentFilter = new IntentFilter();
        intentFilter.addAction("proxy.app.PROXY_PROGRESS");
        intentFilter.addAction("proxy.app.PROXY_REQUEST_CREDENTIAL");
        intentFilter.addAction("proxy.app.PROXY_NOT_IITK");
        intentFilter.addAction("proxy.app.PROXY_LIVE_SESSION");
        return intentFilter;
    }
}

File: app/src/main/java/com/iitk/proxyLogin/activity/SplashActivity.kt
package com.iitk.proxyLogin.activity

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.support.v4.content.LocalBroadcastManager
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.TextView
import com.iitk.proxyLogin.R
import com.iitk.proxyLogin.service.ProxyService
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by kshivang on 22/01/17.

 */

class SplashActivity : AppCompatActivity() {

    private var isAnimationFinished = false
    private var tvProgress: TextView? = null
    private var localBroadcastManager: LocalBroadcastManager? = null

    private val proxyReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            if ("proxy.app.PROXY_PROGRESS" == intent.action) {
                tvProgress!!.text = intent.getStringExtra("Progress")
            } else if ("proxy.app.PROXY_REQUEST_CREDENTIAL" == intent.action) {
                val type = intent.getStringExtra("Type")
                tvProgress!!.text = "You are on IITK $type network"
                onFetched(true)
            } else if ("proxy.app.PROXY_LIVE_SESSION" == intent.action) {
                val lastLogin = intent.getLongExtra("Time",
                        System.currentTimeMillis())
                tvProgress!!.text = "IITK Fortinet refreshed at " + SimpleDateFormat("HH:mm a", Locale.ENGLISH)
                        .format(Date(lastLogin))
                onFetched(false)
            } else {
                tvProgress!!.text = "You are not on IITK network"
                onFetched(false)
            }
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        localBroadcastManager = LocalBroadcastManager.getInstance(this)
        localBroadcastManager!!.registerReceiver(proxyReceiver, makeProxyUpdatesIntentFilter())

        fetchType()

        val bounceAnim = AnimationUtils
                .loadAnimation(applicationContext, R.anim.bounce)

        tvProgress = findViewById<TextView>(R.id.progress_text)
        val tvTitle = findViewById<TextView>(R.id.appName)

        tvTitle.animation = bounceAnim

        bounceAnim.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {}

            override fun onAnimationEnd(animation: Animation) {
                isAnimationFinished = true
            }

            override fun onAnimationRepeat(animation: Animation) {

            }
        })
    }

    private fun fetchType() {
        val proxyServiceIntent = Intent(this, ProxyService::class.java)
        proxyServiceIntent.action = "proxy.service.NETWORK_TYPE"
        startService(proxyServiceIntent)
    }

    private fun onFetched(isLoginRequired: Boolean) {
        startActivity(Intent(this@SplashActivity, if (isLoginRequired)
            LoginActivity::class.java
        else
            SessionActivity::class.java))
    }

    override fun onPause() {
        super.onPause()
        if (this.localBroadcastManager != null) {
            localBroadcastManager!!.unregisterReceiver(proxyReceiver)
        }
    }

    override fun onResume() {
        super.onResume()
        localBroadcastManager!!.registerReceiver(proxyReceiver, makeProxyUpdatesIntentFilter())
        if (isAnimationFinished) {
            finish()
        }
    }

    companion object {


        fun makeProxyUpdatesIntentFilter(): IntentFilter {
            val intentFilter = IntentFilter()
            intentFilter.addAction("proxy.app.PROXY_PROGRESS")
            intentFilter.addAction("proxy.app.PROXY_REQUEST_CREDENTIAL")
            intentFilter.addAction("proxy.app.PROXY_NOT_IITK")
            intentFilter.addAction("proxy.app.PROXY_LIVE_SESSION")
            intentFilter.addAction("proxy.app.NO_WIFI")
            return intentFilter
        }
    }
}
